airsonic-main/src/main/java/org/airsonic/player/service/search/IndexManager.java

```diff
@@ -232,4 +261,79 @@ public void release(IndexType indexType, IndexSearcher indexSearcher) {
+     * Legacy type indexes (files or directories starting with lucene) are deleted.
+     * If there is no index directory, initialize the directory.
+     * If the index directory exists and is not the current version,
+     * initialize the directory.
+     */
+    public void deleteOldIndexFiles() {
+
+        // Delete legacy files unconditionally
+        Arrays.stream(SettingsService.getAirsonicHome()
+                .listFiles((file, name) -> Pattern.compile("^lucene\\d+$").matcher(name).matches())).forEach(old -> {
+                    if (FileUtil.exists(old)) {
+                        LOG.info("Found old index file. Try to delete : {}", old.getAbsolutePath());
+                        try {
+                            if (old.isFile()) {
+                                FileUtils.deleteQuietly(old);
+                            } else {
+                                FileUtils.deleteDirectory(old);
+                            }
+                        } catch (IOException e) {
+                            // Log only if failed
+                            LOG.warn("Failed to delete the old Index : ".concat(old.getAbsolutePath()), e);
+                        }
+                    }
+                });
+
+        // Check if Index is current version
+        if (indexDirectory.get().exists()) {
+
+            if (indexVersionFile.get().exists()) {
+
+                // Index of current version already exists
+                LOG.info("Index was found (index version {}). ", INDEX_VERSION);
+
+            } else {
+
+                // Delete if not current version and create current version directory
+                try {
+                    FileUtils.deleteDirectory(indexDirectory.get());
+                } catch (IOException e) {
+                    LOG.warn("Failed to delete the old Index : ", indexDirectory.get().getAbsolutePath());
+                }
+
+                initializeIndexDirectory();
+
+            }
+        } else {
+
+            // Create current version directory at first start without index
+            initializeIndexDirectory();
+
+        }
+
+    }
+
+    /**
+     * Create index directory for initialization.
+     * Create a directory with a fixed name,
+     * and immediately below create an index version file that represents the current version.
+     */
+    private void initializeIndexDirectory() {
+        try {
+            if (indexDirectory.get().mkdir() && indexVersionFile.get().createNewFile()) {
+                LOG.info("Index directory was created (index version {}). ", INDEX_VERSION);
+            } else {
+                LOG.warn("Failed to create index directory :  (index version {}). ", INDEX_VERSION);
+            }
+        } catch (IOException e) {
+            LOG.error("Failed to create index directory :  (index version "
+                    .concat(Integer.toString(INDEX_VERSION))
+                    .concat("). "), e);
+        }
+    }
+
```

Comment:
Unneeded whitespace

Reply:
Rebased the latest master and changed it to pass all CheckStyle.